keras_cv/layers/preprocessing3d/crop_objects.py

```diff
@@ -0,0 +1,198 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing3d import base_augmentation_layer_3d
+from keras_cv.ops import iou_3d
+from keras_cv.ops.point_cloud import is_within_box3d
+
+POINT_CLOUDS = base_augmentation_layer_3d.POINT_CLOUDS
+BOUNDING_BOXES = base_augmentation_layer_3d.BOUNDING_BOXES
+OBJECT_POINT_CLOUDS = base_augmentation_layer_3d.OBJECT_POINT_CLOUDS
+OBJECT_BOUNDING_BOXES = base_augmentation_layer_3d.OBJECT_BOUNDING_BOXES
+
+
+class CropObjects(base_augmentation_layer_3d.BaseAugmentationLayer3D):
+    """A preprocessing layer which crops objects from point clouds during training.
+
+    This layer will crop points based on object bounding boxes and generates OBJECT_POINT_CLOUDS and OBJECT_BOUNDING_BOXES tensors.
+    During inference time, the output will be identical to input. Call the layer with `training=True` to crop the input points.
```

Comment:
given that during training time, this changes from point_cloud to object_point_cloud, I assume the layers that follow this layer will need to accept object_point_cloud. In that case, does returning point_cloud during inference time make sense?

Reply:
The following layers will accept both object_point_cloud and the original point_cloud. I think we need to return point_cloud as well

Reply:
No augmentation at all during inference.